be-java/src/commonMain/kotlin/lang/temper/be/java/RunJava.kt

```diff
@@ -151,8 +153,22 @@ internal fun runJavaBestEffort(
             false -> when (runJava) {
                 is RunByExec -> {
                     // Currently always for a specific library.
-                    val libDir = relativeOutputDirectoryForLibrary(specifics.backendId, runLibrary!!)
-                    listOf(ToolchainResult(libraryName = runLibrary, doRunJava(libDir.resolve(runDir))))
+                    val workingDir = dirPath(specifics.backendId.uniqueId)
+                    val compileResult = doRunJava(workingDir)
+                    if (compileResult.failure != null) {
+                        return listOf(ToolchainResult(libraryName = runLibrary, compileResult))
+                    }
+                    // TODO Does this ensure a proper java version? If not, do so?
+                    // TODO Minimally, we expect the compile above to be on the right version.
+                    // TODO Functional tests also run entirely on maven.
+                    val java = this[factory.specifics.javaTool]
+                    val classpath = libraryNames.joinToString(cliEnv.pathSeparator) { "$it/target/classes" }
```

Comment:
This copies the same style of classpath gathering as for jshell. Maybe should be more precise in the future to gather only the dependencies of the running library, though.